src/dialogs/ShuffleDialog.java

```diff
@@ -0,0 +1,394 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ *
+ */
+
+/**
+ *
+ * @author S630690
+ */
+final public class ShuffleDialog extends javax.swing.JDialog implements AppFrame {
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1L;
+    
+    private ArrayManager ArrayManager;
+    private JFrame Frame;
+    private UtilFrame UtilFrame;
+
+    private boolean bypassEvents;
+    
+    /**
+     * Creates new form SortPrompt
+     */
+    @SuppressWarnings("unchecked")
+    public ShuffleDialog(ArrayManager ArrayManager, JFrame frame, UtilFrame utilFrame) {
+        super(frame, "ArrayV Advanced Shuffle Editor", true);
+
+        this.ArrayManager = ArrayManager;
+        this.Frame = frame;
+        this.UtilFrame = utilFrame;
+        
+        initComponents();
+
+        bypassEvents = true;
+        this.shuffleEditor.graph = ArrayManager.getShuffle();
+        jList4.setListData(ArrayManager.getDistributionIDs());
+        for(int i = 0; i < ArrayManager.getDistributions().length; i++) {
+            if(ArrayManager.getDistribution().equals(ArrayManager.getDistributions()[i])) {
+                jList4.setSelectedIndex(i);
+                break;
+            }
+        }
+        jList1.setListData(ArrayManager.getDistributionIDs());
+        jList3.setListData(ArrayManager.getDistributionIDs());
+        jList2.setListData(ArrayManager.getShuffleIDs());
+        bypassEvents = false;
+
+        setMinimumSize(new Dimension(765, 300));
+        setAlwaysOnTop(false);
+        reposition();
+        setVisible(true); 
+    }
+
+    @Override
+    public void reposition() {
+    }
+
+    /**
+     * This method is called from within the constructor to initialize the form.
+     * WARNING: Do NOT modify this code. The content of this method is always
+     * regenerated by the Form Editor.
+     */
+    @SuppressWarnings({ "rawtypes" })
+    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
+    private void initComponents() {
+
+        this.shuffleEditor = new ShufflePanel();
+
+        this.jButton1 = new javax.swing.JButton();
+        this.jButton2 = new javax.swing.JButton();
+
+        this.jScrollPane4 = new javax.swing.JScrollPane();
+        this.jList4 = new javax.swing.JList();
+        this.jLabel4 = new javax.swing.JLabel();
+
+        this.jScrollPane1 = new javax.swing.JScrollPane();
+        this.jList1 = new javax.swing.JList();
+        this.jLabel1 = new javax.swing.JLabel();
+
+        this.jScrollPane3 = new javax.swing.JScrollPane();
+        this.jList3 = new javax.swing.JList();
+        this.jLabel3 = new javax.swing.JLabel();
+
+        this.jScrollPane2 = new javax.swing.JScrollPane();
+        this.jList2 = new javax.swing.JList();
+        this.jLabel2 = new javax.swing.JLabel();
```

Comment:
why not fix the numbering?

Reply:
/shrug

Reply:
speaking of which, why do we even number them instead of giving them descriptive names lol

Reply:
yeah, though imo changing it is perfectly fine (since the numbering itself doesn't really provide much benefit between dialogs)

(also, the comments you copied weren't updated properly lol)

Reply:
true, that makes much more sense